### DISCUSSION TITLE
Include mesh in averaging files using the tavg plugin

### DISCUSSION THREAD

**Discussion by dlwldud99** [Nek5000/nekRS#538] [Q&A] [answered]
Hello!

I'm wondering how I can include mesh data in the field files using the tavg plugin.
With the following lines of code:
```
#include "plugins/tavg.hpp"

tavg::setup(nrs);

tavg::run(time);
```
only the first field files of the averaging files (avg,rms,rm2) include the mesh.

I tried setting
`platform->options.setArgs("CHECKPOINT OUTPUT MESH", "TRUE");  `
in UDF_Setup, but had no luck.

Any advice would be appreciated, thank you.

Joy

**Comment by stgeke** [comment, marked as answer]
Instead of just calling `tavg::outfld` use `tavg::outfld(true, true)`

**Reply by dlwldud99** [reply]
Thank you very much!
